pkg/config/validation/validation.go

```diff
@@ -1936,6 +1937,27 @@ func ValidateControlPlaneAuthPolicy(policy meshconfig.AuthenticationPolicy) erro
 	return fmt.Errorf("unrecognized control plane auth policy %q", policy)
 }
 
+func validatePolicyTargetReference(selector *type_beta.WorkloadSelector, targetRef *type_beta.PolicyTargetReference) (v Validation) {
+	if targetRef == nil {
+		return
+	}
+	if selector != nil && targetRef != nil {
+		v = appendErrorf(v, "targetRef and selector cannot both be set")
+	}
```

Comment:
I would suggest we move out this check to keep validatePolicyTargetReference focused

Reply:
I believe we need this check. It was [agreed upon](https://github.com/istio/api/pull/2888#discussion_r1298759828) as a viable alternative to having one-of in the proto. We can just validate that only one of selector or targetRef is set here. 

But maybe I'm missing some more Istio context around existing validation mechanisms. Without that being here, how/where would you propose that we validate that only one of selector or targetRef is set?

Reply:
Of course, we need to check oneof semantics, i mean we can check that out of validateTargetRef. By this we can keep validateTargetRef reusable

Reply:
> i mean we can check that out of validateTargetRef. By this we can keep validateTargetRef reusable

+1